### PR TITLE
add metric for largest segment size

### PR DESCRIPTION
**Pull request by jadami10**
## Description

This adds a metric for the largest segment size on disk by table
- it only publishes the metric if there is at least 1 segment
- the gauge is not reset if all segments are deleted for example. this should be very rare
- this includes the entire segment directory including indices, not just the `columns.psf` file

I tested by running the batch quickstart with a breakpoint and asserting the value at debug time equalled what I saw on disk

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes

This adds a metric for the largest segment size on disk in a server. You can use this to track the growth of you largest segments.

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java [hunk 1]
```diff
@@ -70,6 +70,8 @@
   @Deprecated // Instead use TABLE_TOTAL_SIZE_ON_SERVER
   OFFLINE_TABLE_ESTIMATED_SIZE("OfflineTableEstimatedSize", false),
 
+  LARGEST_SEGMENT_SIZE_ON_DISK("LargestSegmentSizeOnDisk", false),
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java:73]:
```suggestion
  LARGEST_SEGMENT_SIZE_ON_SERVER("LargestSegmentSizeOnServer", false),
```

**Review by Jackie-Jiang — COMMENTED** [review]
Can you please add the prometheus config and some tests? You may refer to #8358 which has very similar changes

File: pinot-controller/src/main/java/org/apache/pinot/controller/util/TableSizeReader.java [hunk 2]
```diff
@@ -99,6 +99,13 @@ public TableSizeDetails getTableSizeDetails(@Nonnull String tableName, @Nonnegat
       _controllerMetrics.setValueOfTableGauge(realtimeTableName, ControllerGauge.TABLE_SIZE_PER_REPLICA_ON_SERVER,
           tableSizeDetails._realtimeSegments._estimatedSizeInBytes / _helixResourceManager.getNumReplicas(
               realtimeTableConfig));
+
+      tableSizeDetails._realtimeSegments._segments.values().stream()
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/util/TableSizeReader.java:103]: (optional) Suggest using the regular non-stream APIs for the following reasons:
1. This is indeed more compact, but slightly hard to read
2. It has worse performance comparing to regular APIs

**Comment by jadami10** [comment]
> Can you please add the prometheus config and some tests?  You may refer to https://github.com/apache/pinot/pull/8358 which has very similar changes

Added both. It tests for realtime, offline, and all segments missing. thank you for the pointer PR

File: pinot-controller/src/test/java/org/apache/pinot/controller/api/TableSizeReaderTest.java [hunk 3]
```diff
@@ -90,15 +90,15 @@ public void setUp()
...
-          String path = (String) invocationOnMock.getArguments()[0];
-          if (path.contains("realtime_REALTIME")) {
-            return TableConfigUtils.toZNRecord(tableConfig);
-          }
-          if (path.contains("offline_OFFLINE")) {
-            return TableConfigUtils.toZNRecord(tableConfig);
-          }
-          return null;
-        });
+      String path = (String) invocationOnMock.getArguments()[0];
```
Comment (jadami10) [inline pinot-controller/src/test/java/org/apache/pinot/controller/api/TableSizeReaderTest.java:93]: not changes here. the reformat changed the whitespace

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**